assets/pages/oneoff-contributions/reducers/reducers.js

```diff
@@ -19,55 +23,61 @@ import type { Action } from '../actions/oneoffContributionsActions';
 
 export type State = {
   amount: number,
-  country: string,
+  currency: Currency,
+  country: IsoCountry,
   error: ?string,
   payPalType: PayPalButtonType,
 };
 
-
-// ----- Setup ----- //
-
-const initialState: State = {
-  amount: 50,
-  country: 'GB',
-  error: null,
-  payPalType: 'NotSet',
+export type CombinedState = {
+  oneoffContrib: State,
+  intCmp: string,
+  user: UserState,
+  stripeCheckout: StripeCheckoutState,
+  csrf: CsrfState,
 };
 
-
 // ----- Reducers ----- //
 
-function oneoffContrib(
-  state: State = initialState,
-  action: Action): State {
+function oneoffContrib(amount: number, currency: Currency, country: IsoCountry) {
+
+  const initialState: State = {
+    amount,
+    currency,
+    country,
+    error: null,
+    payPalType: 'NotSet',
+  };
 
-  switch (action.type) {
+  return (state: State = initialState, action: Action): State => {
 
-    case 'SET_CONTRIB_VALUE':
-      return Object.assign({}, state, { amount: action.value });
+    switch (action.type) {
 
-    case 'CHECKOUT_ERROR':
-      return Object.assign({}, state, { error: action.message });
+      case 'SET_CONTRIB_VALUE':
+        return Object.assign({}, state, { amount: action.value });
 
-    case 'SET_PAYPAL_BUTTON' :
-      return Object.assign({}, state, { payPalType: action.value });
+      case 'CHECKOUT_ERROR':
+        return Object.assign({}, state, { error: action.message });
 
-    default:
-      return state;
+      case 'SET_PAYPAL_BUTTON' :
+        return Object.assign({}, state, { payPalType: action.value });
 
-  }
+      default:
+        return state;
 
+    }
+  };
 }
 
 
 // ----- Exports ----- //
 
-export default combineReducers({
-  oneoffContrib,
-  intCmp,
-  user,
-  stripeCheckout,
-  payPalExpressCheckout,
-  payPalContributionsCheckout,
-  csrf,
-});
+export default (amount: number, currency: Currency, country: IsoCountry) =>
+  combineReducers({
```

Comment:
The function is called `createRootOneOffContribReducer` and it has a single line - I think it is reasonable to assume it returns a `rootOneOffContribReducer` without adding another variable name